### PR TITLE
[CU-860rh2p6u]: sort weapons

### PR DESCRIPTION
**Pull request by FlavianEng**
## Changelog:  
- Sort weapons by custom name order
- Remove useless log

## Checklist before requesting a review
-   [x] I have performed a self-review of my code
-   [x] My pull request title respect the naming convention: [**_CU-{ClickUp Id}_**] **_{summary}_**

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-860rh2p6u US: As a player, I'd like to have the weapons always in the same order](https://app.clickup.com/t/860rh2p6u)

**Comment by MASBERNATQ** [comment]
Envoyer les settings dans l'événement `start-placing-boats` afin de permettre à l'hôte d'avoir les armes de la partie dans le bon ordre.

File: src/engine/service/game-arsenal.service.spec.ts [hunk 1]
```diff
@@ -93,4 +93,50 @@ describe('GameArsenalService', () => {
...
+    it('should return the complete ordered array of weapons enum', () => {
+      expect(
+        service.sortWeapons([
+          WeaponNameEnum.ULTIMATE,
+          WeaponNameEnum.CROSS,
+          WeaponNameEnum.TRIPLE,
+          WeaponNameEnum.CLASSIC,
+          WeaponNameEnum.RANDOM,
+        ]),
+      ).toEqual([...WEAPONS_DISPLAY_ORDER]);
```
[the remark this answers is not in the document]
Comment (FlavianEng) [reply src/engine/service/game-arsenal.service.spec.ts:107]: oui

[the remark this answers is not in the document]
Comment (FlavianEng) [reply src/engine/service/game-arsenal.service.spec.ts:107]: Cause readonly : https://github.com/KinderCorp/battleship-back/blob/f2f24022b787be64b6d7930346b8d9b0b41f5b24/src/engine/game-instance.const.ts#L15

**Review by MASBERNATQ — APPROVED** [review]

**Pull request merged**